conda_smithy/templates/github-actions.yml.tmpl

```diff
@@ -38,66 +78,34 @@ jobs:
             SHORT_CONFIG: {{ data.short_config_name }}
         {%- endif %}
             UPLOAD_PACKAGES: {{ data.upload }}
-        {%- if data.build_platform.startswith("osx-64") %}
-            os: macos
-        {%- if github_actions.self_hosted %}
-            labels: ['macOS', 'self-hosted', 'x64']
-        {%- endif %}
-        {%- elif data.build_platform.startswith("osx-arm64") %}
-            os: macos
+            os: {{ platform_vars[data.build_platform]['os'] }}
+        {#- namespace needed to handle jinja scoping rules #}
+        {%- set ns = namespace(gpu=False) %}
         {%- if github_actions.self_hosted %}
-            labels: ['macOS', 'self-hosted', 'ARM64']
+            labels:
+              - self-hosted
+              - {{ platform_vars[data.build_platform]['self_hosted_label_os'] }}
+              {%- if platform_vars[data.build_platform]['self_hosted_label_arch'] %}
+              - {{ platform_vars[data.build_platform]['self_hosted_label_arch'] }}
+              {%- endif %}
+              {%- for label in github_actions.self_hosted_labels %}
+                {%- if 'gpu' in label %}
+                  {%- set ns.gpu = True %}
```

Comment:
We are using `ns` as a global namespace to make sure that the line below for `CONDA_FORGE_DOCKER_RUN_ARGS` can see this flag. Otherwise, scoping rules will prevent it.